crates/sui-graphql-client/src/lib.rs

```diff
@@ -376,53 +379,57 @@ impl Client {
     // ===========================================================================
 
     /// Get the chain identifier.
-    pub async fn chain_id(&self) -> Result<String, Error> {
+    pub async fn chain_id(&self) -> Result<String, error::Error> {
         let operation = ChainIdentifierQuery::build(());
         let response = self.run_query(&operation).await?;
 
         if let Some(errors) = response.errors {
-            return Err(Error::msg(format!("{:?}", errors)));
+            return Err(QueryResponseError(errors).into());
         }
 
         response
             .data
             .map(|e| e.chain_identifier)
-            .ok_or_else(|| Error::msg("No data in response"))
+            .ok_or_else(|| EmptyResponse.into())
```

Comment:
We probably don't need EmptyResponse as a separate error Kind from a Deserialization Error (once we define the error kinds) but may want something like this type or a string that just says the response was empty